public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -145,19 +146,9 @@ export default class PromQlLanguageProvider extends LanguageProvider {
   };
 
   async loadMetricsMetadata() {
-    // The metadata endpoint is experimental and
-    // we have customers who are not implementing it.
-    // This is to be a temporary fix until the Observability Metrics Squad
-    // has time to implement the 2022Q3 plan
-    // to detect prometheus versions and implementations.
-    // This will allow us to see if endpoints such as api/v1/metadata
-    // are implemented or not and handle them as such
-    this.metricsMetadata = {};
-
-    // PREVIOUS IMPLEMENTATION FOR NOTES
-    // this.metricsMetadata = fixSummariesMetadata(
-    //   await this.request('/api/v1/metadata', {}, {}, { showErrorAlert: false })
-    // );
+    this.metricsMetadata = fixSummariesMetadata(
+      await this.request('/api/v1/metadata', {}, {}, { showErrorAlert: false })
+    );
```

Comment:
What do you think about adding some error handling here? Some `try catch` or similar approaches to handle errors with these requests.

Reply:
@obetomuniz this is the request method implementation: 
https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/language_provider.ts#L123-L132
If there is an error it will return the default value provided here. In that case it will be `{}`

Reply:
Sure thing @itsmylife. Also, it seems `fixSummariesMetadata` deals well with `{}`. Makes sense. Thanks for the feedback and clarification.